assets/agenda/components/AgendaApp.jsx

```diff
@@ -85,8 +85,15 @@ class AgendaApp extends BaseApp {
     }
 
     render() {
-        if (this.state.initialLoad) {
-            return this.renderLoader();
+        if (this.state.initialLoad) {    
+            if(this.props.userType !== 'administrator') {
```

Comment:
I think it should also check if user has any products, otherwise this will be there for every non admin user no?
and this should be in theory only after that initialLoad block:
```ts
if (this.state.initialLoad) {
    return this.renderLoader();
}
if (this user is not admin && this user has no products) {
    return <div>{'You don't have any products...'})</div>;
}
...
```

Reply:
that's not worked, I guess because we got `this.state.initialLoad = true`, so it returns only `renderLoader()`

Reply:
you mean that it keeps that state `initialLoad` because of missing products? I think we should handle that then, there is probably an error returned from the server so we can handle that in the action and change the state accordingly